### DISCUSSION TITLE
How do I get what is sent by webhook events

### DISCUSSION THREAD

**Discussion by AdonisPat** [python-gitlab/python-gitlab#2108] [Q&A]
good evening

I tell you that I am trying to retrieve information that is sent by the webhook events, for example from the pipeline and I am not finding it in the documentation, how to do it?

I wanted to know if there is any easy way to get them back.

Context:
* In the group of my project, configure the webhook events, for example the Pipeline events and this sends information to a kafka.
* I want to do a bulk first ingest, so I'm going through all the projects that have webhooks configured and retrieving the information, as sent by the webhook pipeline events, but I realize there's nothing in the library that I have easily instead I was thinking of retrieving data by data and putting together the json. Is this the case or is there another way to get this information?

I wait an answer.

Thanks and regards

**Comment by nejch** [comment]
@AdonisPat - is there a specific upstream API endpoint listed in https://docs.gitlab.com/ee/api/api_resources.html  that you are looking for to be implemented in the library?

**Reply by AdonisPat** [reply]
Thanks for your answer.
Actually I am looking to retrieve the information that is provided through this -> https://docs.gitlab.com/ee/user/project/integrations/webhook_events.html#pipeline-events

And I don't know how to get it back.

**Reply by AdonisPat** [reply]
I have the event pipeline_events set to true and configured with an endpoint But I want to retrieve all the pipeline events generated so far.

**Reply by AdonisPat** [reply]
I do not want to recover the information of the event configuration itself, but what I want to recover is all the data of all the events already generated previously

**Reply by nejch** [reply]
I see. If I understand correctly, this is the actual payload sent in the webhook, so this is only something you can retrieve on the webhook receiver end, not something this library (or any GitLab API client library for that matter) was built for.

What you can do with python-gitlab is configure the actual webhooks and retrieve their configuration as you say above. But you'll need to have some kind of webhook sink to collect the actual hooks fired and dispatch them further or so.

If GitLab implements an API endpoint to, for example "list details for webhooks fired on this configured project hook", then this is something we can add here. You can maybe open an upstream issue on GitLab.com for that.

**Reply by AdonisPat** [reply]
Taking into account that when I retrieve information from the pipeline with the library, it brings me certain information in the json and when I compare it with the json sent from the pipeline event, the latter has many more fields

**Reply by nejch** [reply]
That's probably by design, to make webhooks convenient and provide everything in a single payload. With the API, you might be able to retrieve more information once you have the pipeline ID by calling the associated endpoints separately (pipeline jobs, commit, merge requests API etc).

**Reply by AdonisPat** [reply]
aa okay, well that was more or less what I was doing but I wanted to make sure. As you say I will have to open a case in gitlab

**Reply by AdonisPat** [reply]
Thank you very much for your response. And congratulations on the library. It helps us a lot.

**Reply by nejch** [reply]
Thanks for the quick feedback!

You could also try the GraphQL API to return everything you need in a single query, though we do not support that yet here unfortunately. And it might not have all the fields that you want supported either.
